chromium_src/components/permissions/permission_context_base.cc

```diff
@@ -63,9 +63,32 @@ void PermissionContextBase::PermissionDecided(const PermissionRequestID& id,
       }
     }
   }
-  PermissionContextBase_ChromiumImpl::PermissionDecided(
-      id, requesting_origin, embedding_origin, content_setting, is_one_time,
-      is_final_decision);
+
+  if (!IsGroupedPermissionType(content_settings_type())) {
+    PermissionContextBase_ChromiumImpl::PermissionDecided(
+        id, requesting_origin, embedding_origin, content_setting, is_one_time,
+        is_final_decision);
+    return;
+  }
+
+  DCHECK(content_setting == CONTENT_SETTING_ALLOW ||
+         content_setting == CONTENT_SETTING_BLOCK ||
+         content_setting == CONTENT_SETTING_DEFAULT);
+  UserMadePermissionDecision(id, requesting_origin, embedding_origin,
+                             content_setting);
+
+  bool persist = content_setting != CONTENT_SETTING_DEFAULT;
+
+  auto grouped_request = pending_grouped_requests_.find(id.ToString());
+  DCHECK(grouped_request != pending_grouped_requests_.end());
+  DCHECK(grouped_request->second);
+
+  auto callback = grouped_request->second->GetNextCallback();
```

Comment:
shouldn't we check `!IsDone()` here? Can you please describe what exactly this change fixes? Maybe extend a commit message to include more details.

Reply:
Added that check and fleshed out the commit message to explain things better here https://github.com/brave/brave-core/pull/16324/commits/279574fdd6180c9ec2831259a2c77e24ca4e9917.

@yrliou May be able to provide additional context here?